vlo-web-app/src/main/java/eu/clarin/cmdi/vlo/service/impl/ExposureTrackerImpl.java

```diff
@@ -43,26 +43,37 @@
 public class ExposureTrackerImpl implements ExposureTracker {
 
     private final static Logger logger = LoggerFactory.getLogger(ExposureTrackerImpl.class);
-    private final QueryFacetsSelection selection;
     private final VloConfig vloConfig;
     
-    public ExposureTrackerImpl(QueryFacetsSelection selection) {
-        this.vloConfig = VloWicketApplication.get().getVloConfig();
-        this.selection = selection;
+    public ExposureTrackerImpl(VloConfig vloConfig) {
+        this.vloConfig = vloConfig;
     }
-
+    
+    private String anonymizeIP(String ip) {
```

Comment:
small suggestion, and to be tested for actual performance but I think this might be done more more efficiently (and less code) via regular expression replacement. The following pattern `([\d\.]+)\.\d+` would probably do, then replace (in case of match) with `matcher.group(1) + ".0"`.

https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html
https://docs.oracle.com/javase/7/docs/api/java/util/regex/Matcher.html#group(int)